MapboxAndroidSDK/src/main/java/com/mapbox/mapboxsdk/overlay/GpsLocationProvider.java

```diff
@@ -58,19 +88,30 @@ public void setLocationUpdateMinDistance(final float meters) {
     public boolean startLocationProvider(UserLocationOverlay myLocationConsumer) {
         mMyLocationConsumer = myLocationConsumer;
         boolean result = false;
-        for (final String provider : mLocationManager.getProviders(true)) {
-            if (LocationManager.GPS_PROVIDER.equals(provider)
-                    || LocationManager.PASSIVE_PROVIDER.equals(provider)
-                    || LocationManager.NETWORK_PROVIDER.equals(provider)) {
-                result = true;
-                if (mLocation == null) {
-                    mLocation = mLocationManager.getLastKnownLocation(provider);
-                    if (mLocation != null) {
-                        mMyLocationConsumer.onLocationChanged(mLocation, this);
+        if(testMode == true) {
```

Comment:
This is one area that the code could be mocked, the else section is entirely skipped and it seems that would be the place that needed to be tested.